### PR TITLE
Reject file to be flashed based on file extension

### PR DESCRIPTION
**Pull request by juhhov**
## Status
**READY**

## Description
Check file extension of a file to be flashed and reject if it is not known.

### CODE REVIEW

**Comment by jupe** [comment]
could you solve conflicts so we can merge in

File: mbed_flasher/common.py [hunk 1]
```diff
@@ -15,11 +15,16 @@
...
+import os
 import time
 import six
 
+from mbed_flasher.return_codes import EXIT_CODE_FILE_MISSING
+from mbed_flasher.return_codes import EXIT_CODE_DAPLINK_USER_ERROR
+
 
 DEFAULT_RETRY_AMOUNT = 3
+ALLOWED_FILE_EXTENSIONS = (".bin", ".hex", ".act", ".cfg")
```
Comment (jonikula) [inline mbed_flasher/common.py]: .elf file is missing. Or have you checked from client team (I believe) if they are still using those?

Comment (jupe) [reply mbed_flasher/common.py]: elf file is not supported for Daplink - based on Russ comments.

Comment (jupe) [reply mbed_flasher/common.py]: double check with local K64F by drag-n-drop elf file - Daplink doesn't do anything with that -> not supported like @c1728p9 mentioned.

Comment (jonikula) [reply mbed_flasher/common.py]: Ok, perhaps they don't use flasher with that then.

Comment (juhhov) [reply mbed_flasher/common.py]: They can't use daplink with .elf, since daplink is the one not supporting it. Pyocd supports .elf, so maybe that.

**Comment by juhhov** [comment]
x86 test results:
```
systest@raasdesktop:~/workspace/mbed-flasher$ coverage run -m unittest discover -s test
.usage: mbedflash [-h] [-v] [-s] <command> ...
mbedflash: error: argument <command>: invalid choice: 'discover' (choose from 'list', 'flashers', 'version', 'flash', 'reset', 'erase')
......[ERROR](mbed-flasher): Flashing failed: The transfer timed out.. tid=0240000034544e45004b00018aa9001d2011000097969900
[ERROR](mbed-flasher): Failed: The transfer timed out.
....................usage: mbedflash [-h] [-v] [-s] <command> ...
mbedflash: error: too few arguments
...usage: mbedflash [-h] [-v] [-s] <command> ...
mbedflash: error: argument <command>: invalid choice: 'discover' (choose from 'list', 'flashers', 'version', 'flash', 'reset', 'erase')
..................usage: mbedflash [-h] [-v] [-s] <command> ...

For specific command help, run: mbedflash <command> --help

optional arguments:
  -h, --help     show this help message and exit
  -v, --verbose  Verbose level... repeat up to three times.
  -s, --silent   Silent - only errors will be printed.

command:
  <command>      command help
    list         Prints a list of supported platforms.
    flashers     Prints a list of supported flashers.
    version      Display version information
    flash        Flash given resource
    reset        Reset given resource
    erase        Erase given resource
.usage: mbedflash [-h] [-v] [-s] <command> ...
mbedflash: error: argument <command>: invalid choice: 'a' (choose from 'list', 'flashers', 'version', 'flash', 'reset', 'erase')
..s.s..................................................................
----------------------------------------------------------------------
Ran 120 tests in 391.717s

OK (skipped=2)
```

**Pull request merged**